src/wgapi_userspace.rs

```diff
@@ -150,11 +150,15 @@ impl WireguardInterfaceApi for WireguardApiUserspace {
     fn assign_address(&self, address: &IpAddrMask) -> Result<(), WireguardInterfaceError> {
         debug!("Assigning address {address} to interface {}", self.ifname);
         let output = if cfg!(target_os = "macos") {
-            // On macOS, interface is point-to-point and requires a pair of addresses
-            let address_string = address.ip.to_string();
-            Command::new("ifconfig")
-                .args([&self.ifname, &address_string, &address_string])
-                .output()?
+          match address.ip {
+              IpAddr::V4(ip) => Command::new("ifconfig")
+                  // On macOS ipv4, interface is point-to-point and requires a pair of addresses
+                  .args([&self.ifname, "inet", &ip.to_string(), &ip.to_string()])
```

Comment:
You should have kept `let address_string = address.ip.to_string();`. Now the conversion is going to be done twice.

Reply:
Good catch!  I believe I corrected this as you are asking for.

Cheers